### ISSUE TITLE
Ensuring I have attention right

### ISSUE THREAD

**Issue by muellerzr** [dreamquark-ai/tabnet#93]
Hi there! Could you please help verify that I've made sure to do attention right? I'm working off the `fastai` implementation, and so it would be faster to read up [here](https://forums.fast.ai/t/tabnet-with-fastai-v2/62600/49) but essentially I made a modification to his model that can return the masks. So it currently looks like so:
```python
learn.model.eval()
for batch_nb, data in enumerate(dl):
  with torch.no_grad():
    out, M_loss, M_explain, masks = learn.model(data[0], data[1], True)
  for key, value in masks.items():
    masks[key] = csc_matrix.dot(value.numpy(), matrix)
  if batch_nb == 0:
    res_explain = csc_matrix.dot(M_explain.numpy(),
                                 matrix)
    res_masks = masks
  else:
    res_explain = np.vstack([res_explain,
                             csc_matrix.dot(M_explain.numpy(),
                                            matrix)])
    for key, value in masks.items():
      res_masks[key] = np.vstack([res_masks[key], value])
```
From here to plot, I do:
```python
fig, axs = plt.subplots(1, 3, figsize=(20,20))
for i in range(3):
  axs[i].imshow(np.expand_dims(res_masks[0][i], 0))
```

Now I chose to do the `np.expand_dims` as it let's us visualize on an indivudal item level what is going on. Is this the correct way to do this sort of analysis? Or should I have included it at a batch level (or does it really not make a difference in the end).

Thanks!

**Comment by muellerzr** [comment]
I'm reopening this as I don't think I'm 100% right after seeing the PR for categorical embeddings, can you read the post on the forums and advise me where to adjust? Thanks!
